digitalpy/core/zmanager/integration_manager.py

```diff
@@ -11,33 +11,50 @@
+        self.integration_manager_puller_port = integration_manager_puller_port
+        self.integration_manager_publisher_protocol = integration_manager_publisher_protocol
+        self.integration_manager_publisher_address = integration_manager_publisher_address
+        self.integration_manager_publisher_port = integration_manager_publisher_port
+
+
     def initialize_connections(self):
         context = zmq.Context()
 
         # create a pull socket
         pull_socket = context.socket(zmq.PULL)
-        pull_socket.setsockopt(zmq.RCVHWM, 1000)  # limit the number of unprocessed requests that can be queued
-        pull_socket.bind("tcp://127.0.0.1:12345")
+        # unlimited as trunkating can result in unsent data and broken messages
+        # TODO: determine a sane default
+        pull_socket.setsockopt(zmq.RCVHWM, 0)
+        pull_socket.bind(f"{self.integration_manager_puller_protocol}://{self.integration_manager_puller_address}:{self.integration_manager_puller_port}")
         self.pull_socket = pull_socket
 
         # create a pub socket
         pub_socket = context.socket(zmq.PUB)
-        pub_socket.setsockopt(zmq.SNDHWM, 1000)  # limit the number of unsent responses that can be queued
-        pub_socket.bind("tcp://127.0.0.1:12346")
+        # unlimited as trunkating can result in unsent data and broken messages
+        # TODO: determine a sane default
+        pub_socket.setsockopt(zmq.SNDHWM, 0)
+        pub_socket.bind(f"{self.integration_manager_publisher_protocol}://{self.integration_manager_publisher_address}:{self.integration_manager_publisher_port}")
         self.pub_socket = pub_socket
 
-    def main(self):
+    def start(self):
+        """this is the main running function for the integration manager"""
         self.initialize_connections()
 
         while True:
-            # receive a message from a client
-            request = self.pull_socket.recv()
-            response_protocol, response_object_unserialized = request.split(b',', 1)
-            subject = b"/"+response_protocol+b"/message"
-            # measure the processing time of the request
-
             try:
-                # send the response back to the client
-                self.pub_socket.send(subject + b" " + response_object_unserialized)
+                # receive a message from a client
+                request = self.pull_socket.recv_multipart()[0]
+                response_protocol, response_object_unserialized = request.split(b',', 1)
+                subject = b"/messages" + response_protocol
+
+                try:
+                    # send the response back to the client
+                    self.pub_socket.send(subject + b" " + response_object_unserialized)
+                except Exception as ex:
+                    print("Error sending response to client: {}".format(ex))
             except Exception as ex:
-                print(ex)
\ No newline at end of file
+                print("Error "+str(ex))
```

Comment:
**[pep8]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>W292 no newline at end of file